pkg/cache/tas_flavor_snapshot.go

```diff
@@ -379,26 +382,133 @@ func (s *TASFlavorSnapshot) Fits(flavorUsage workload.TASFlavorUsage) bool {
 // the TAS requests in the flavor handled by the snapshot.
 // The simulateEmpty parameter allows to look for the assignment under the
 // assumption that all TAS workloads are preempted.
-func (s *TASFlavorSnapshot) FindTopologyAssignmentsForFlavor(flavorTASRequests FlavorTASRequests, simulateEmpty bool) TASAssignmentsResult {
+func (s *TASFlavorSnapshot) FindTopologyAssignmentsForFlavor(flavorTASRequests FlavorTASRequests, simulateEmpty bool, wl *kueue.Workload) TASAssignmentsResult {
 	result := make(map[kueue.PodSetReference]tasPodSetAssignmentResult)
 	assumedUsage := make(map[utiltas.TopologyDomainID]resources.Requests)
 	for _, tr := range flavorTASRequests {
-		assignment, reason := s.findTopologyAssignment(tr, assumedUsage, simulateEmpty)
-		result[tr.PodSet.Name] = tasPodSetAssignmentResult{TopologyAssignment: assignment, FailureReason: reason}
-		if reason != "" {
-			return result
-		}
-		for _, domain := range assignment.Domains {
-			domainID := utiltas.DomainID(domain.Values)
-			if assumedUsage[domainID] == nil {
-				assumedUsage[domainID] = resources.Requests{}
+		if workload.HasNodeToReplace(wl) {
+			// In case of looking for replacement TopologyRequest has only
```

Comment:
please pull this out to a function

Reply:
Could you elaborate please?

Reply:
Something like, just to make the for-loop more readable/smaller

```
if workload.HasNodeToReplace(wl) {
  assignment, ok = assignmentReplacingNode(args)
  continue
}
```

Reply:
Improved, please see if this aligns with your suggestion